python/hail/dataset.py

```diff
@@ -750,6 +750,63 @@ def annotate_variants_expr(self, condition):
         pargs = ['annotatevariants', 'expr', '-c', condition]
         return self.hc.run_command(self, pargs)
 
+    def annotate_variants_keytable(self, keytable, condition, vdsKey=None):
```

Comment:
I'll fix the camel-case.

I envisioned `vds_key` as a Hail expression evaluating to a structure whose fields were the join-keys. Do you prefer the more limited interface of sample annotation paths?